src/main/java/com/mathworks/ci/pipeline/MatlabBuildStepExecution.java

```diff
@@ -0,0 +1,60 @@
+package com.mathworks.ci.pipeline;
+
+/**
+ * Copyright 2022-2024 The MathWorks, Inc.
+ *  
+ */
+
+import java.io.IOException;
+
+import org.jenkinsci.plugins.workflow.steps.StepContext;
+import org.jenkinsci.plugins.workflow.steps.SynchronousNonBlockingStepExecution;
+import hudson.model.Result;
+
+import com.mathworks.ci.actions.MatlabActionFactory;
+import com.mathworks.ci.actions.RunMatlabBuildAction;
+import com.mathworks.ci.parameters.BuildActionParameters;
+
+public class MatlabBuildStepExecution extends SynchronousNonBlockingStepExecution<Void> {
+    
+    private static final long serialVersionUID = 4771831219402275744L;
+
+    private MatlabActionFactory factory;
+    private RunMatlabBuildStep step;
+
+    public MatlabBuildStepExecution(MatlabActionFactory factory, StepContext ctx, RunMatlabBuildStep step) throws IOException, InterruptedException {
+        super(ctx);
+
+        this.factory = factory;
+        this.step = step;
+    }
+    
+    public MatlabBuildStepExecution(StepContext ctx, RunMatlabBuildStep step) throws IOException, InterruptedException {
+        this(new MatlabActionFactory(), ctx, step);
+    }
+
+    @Override
+    public Void run() throws Exception {
+        BuildActionParameters params = new BuildActionParameters(
```

Comment:
Yes, this is an issue when you use scripted pipeline especially with node {} block. although the updated document mentions that it allocates Workspace I did not see that happening. I guess the script throws error if it does not find Workspace. so I am ok if works fine with the scripted pipeline.

Reply:
I'm still a little bit confused as to what you mean. Do you mean that in the past the `node` step has not allocated a workspace in your experience but it seems like they have updated it to allocate one?

If so, judging from [this](https://groups.google.com/g/jenkinsci-users/c/513qLiYlXHc) thread, my guess would be that that was a bug rather than an intended feature. Probably worth ensuring that we can fail gracefully in that case however.

Reply:
Yes same error but i had seen it if I use  node{}

Reply:
Gotcha! It seems like other popular plugins are going ahead and creating the workspace if it doesn't exist, so I will make the fix that you recommended off the bat.

I'll probably add it to `MatlabCommandRunner` so that the behavior is the same across all steps if that makes sense?